src/scikit_build_core/settings/metadata.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+import pathlib
+import warnings
+
+from pyproject_metadata import StandardMetadata
+
+from .._compat import importlib, tomllib
+from ..settings.skbuild_model import ScikitBuildSettings
+
+__all__ = ["setuptools_scm_version", "fancy_pypi_readme", "get_standard_metadata"]
+
+
+def __dir__() -> list[str]:
+    return __all__
+
+
+def setuptools_scm_version(pyproject_path: pathlib.Path) -> str:
+    from setuptools_scm import Configuration, _get_version
+
+    config = Configuration.from_file(str(pyproject_path))
+    version: str = _get_version(config)
+
+    return version
+
+
+def fancy_pypi_readme(pyproject_path: pathlib.Path) -> str | dict[str, str | None]:
+    from hatch_fancy_pypi_readme._builder import build_text
+    from hatch_fancy_pypi_readme._config import load_and_validate_config
+
+    with pyproject_path.open("rb") as ft:
+        pyproject = tomllib.load(ft)
+
+    config = load_and_validate_config(
+        pyproject["tool"]["hatch"]["metadata"]["hooks"]["fancy-pypi-readme"]
+    )
+
+    return {
+        "content-type": config.content_type,
+        "text": build_text(config.fragments, config.substitutions),
+    }
```

Comment:
@hynek, would it be possible to expose this as a public function so we don't have to dip into internals? Happy to contribute it if so.

Reply:
Do we want to decide on a signature for the plugin function first? For example, I proposed to pass the pyproject path as a reasonable approximation of the project root as well as allowing it to be loaded for its own config, but maybe passing the project root as a string plus the already parsed pyproject contents makes more sense as it could conceivably not be at the project root sometimes (perhaps not in the same folder as .git anyway, for setuptools_scm), plus it saves having to reparse the pyproject contents which we had to do to find these plugins anyway. But having the pyproject dict already would mean a different signature required from the underlying packages.

Reply:
Yes, passing a parsed pyproject.toml seems cleaner. That would be easy here (you'd just pick out `["tool"]["hatch"]["metadata"]["hooks"]["fancy-pypi-readme"]`). For setuptools_scm, you'd need to build the Configuration from a parsed pyproject.toml, but maybe the public API coming soon makes that possible?

Reply:
My thought here was that some tools might not want to use pyproject.toml, for example your conjectured CMake based plugin would probably want to know where the top level CMakeLists.txt was instead. I felt that passing the path to pyproject.toml meant that people could get the root directory of the project as the parent of that path, which they could not do from the contents. I wondered about passing both the project root and the contents, which complicates the signature but saves reparsing pyproject.toml. However, it seems like there is an implicit assumption that build_wheel will only ever be called with the current working directory set to the project root, so plugins could probably just use e.g. `Path("CMakeLists.txt")` anyway. Unless anyone knows any reason not to do this, then I guess we can just pass the pyproject_dict by itself.

Reply:
The pieces of information that might be useful: root directory (as you said, that's the current directory), the pyproject.toml info (though this could be read from the root directory, it's probably easy to pass and saves reparsing), and _maybe_ the config settings.

Reply:
Sorry I'm extremely busy for the foreseeable future:

![image](https://user-images.githubusercontent.com/41240/219295632-a759d9df-466a-45f0-bbdb-6db266c99411.jpeg)

But feel free to meanwhile open an issue on my tracker with an outline what and why you'd like!